modules/hostap/src/supp_api.c

```diff
@@ -144,6 +144,73 @@ static void supp_shell_connect_status(struct k_work *work)
 	k_mutex_unlock(&wpa_supplicant_mutex);
 }
 
+static int wpa_supp_supported_channels(struct wpa_supplicant *wpa_s, uint8_t band, char **chan_list)
+{
+	struct hostapd_hw_modes *mode = NULL;
+	int i, j;
+	int offset, retval;
+	int size;
+	char *_chan_list;
+
+	if ((band & BIT(WIFI_FREQ_BAND_2_4_GHZ)) == BIT(WIFI_FREQ_BAND_2_4_GHZ)) {
+		mode = NULL;
+		for (j = 0; j < ((wpa_s)->hw).num_modes; j++) {
+			if (((wpa_s)->hw).modes[j].mode != HOSTAPD_MODE_IEEE80211A) {
+				mode = &((wpa_s)->hw).modes[j];
+				break;
+			}
+		}
+
+		if (mode) {
+			size = ((mode->num_channels) * 6) + 1;
+			_chan_list = (char *)k_malloc(size);
+			if (!_chan_list) {
+				wpa_printf(MSG_ERROR, "Mem alloc failed for 2.4GHz channel list\n");
+				return -ENOMEM;
+			}
+
+			retval = 0;
+			offset = 0;
+			for (i = 0; i < mode->num_channels; i++) {
+				retval = snprintf(_chan_list + offset, 6, " %d",
+						  mode->channels[i].freq);
+				offset += retval;
+			}
+			*chan_list = _chan_list;
+		}
```

Comment:
This logic is common for 2.4/5, so, can be refactored?

Reply:
Done